owltorrent-network/src/main/java/edu/rice/owltorrent/network/TorrentManager.java

```diff
@@ -374,6 +357,15 @@ public void reportBlockFailed(FileBlockInfo blockInfo) {
     status.status.get(blockInfo.getOffsetWithinPiece() / status.blockLength).set(BLOCK_NOT_STARTED);
   }
 
+  private int getBlockLength(PieceStatus pieceStatus, int blockIndex) {
+    if (isLastBlock(pieceStatus, blockIndex)
+        && ((int) torrent.getLastPieceLength()) % pieceStatus.blockLength != 0) {
```

Comment:
Is it possible in our code for the piece length not to be a multiple of block size? If so this will be wrong for the last block in a given piece.

Reply:
So if it is the last block and the last piece length is a multiple of block size, that means the last block size is equal to the default block size.